pkg/internal/converter/convert.go

```diff
@@ -253,6 +253,11 @@ func Convert(o Options, pathOptions *clientcmd.PathOptions) error {
 		}
 
 		switch o.TokenOptions.LoginMethod {
+		case token.AzureDeveloperCLILogin:
```

Comment:
please add a unit test

Reply:
@weinong Added a couple valid use cases for `azd`.  I did not add all the legazy or override use cases since there should not be any special support required for these.

If you would prefer me to add them just let me know.